utility/jsonToJdl.js

```diff
@@ -0,0 +1,268 @@
+            if(communications[i].clientName !== "" && communications[i].serverName !== "") {
+            const data = `
+communication {
+    client "${communications[i].clientName.toLowerCase()}",
+    server "${communications[i].serverName.toLowerCase()}"
+}
+
+`;
+            communicationData.push(data);
+            }       
+        }
+    }
+
+    // Deployment 
+    const deployment = jsonData.deployment;
+    var deploymentData;
+    if (deployment !== undefined) {
+        // Error handling
+        let deploymentError = [];
+        if (deployment.cloudProvider === undefined || deployment.cloudProvider === "") {
+            deploymentError.push("Cloud provider cannot be empty");
+        }
+        if (deployment.cloudProvider !== undefined) {
+            if (deployment.cloudProvider === "aws") {
+                if(deployment.awsAccountId === undefined || deployment.awsAccountId === ""){
+                    deploymentError.push("AWS account id cannot be empty");
+                }
+                if(deployment.awsRegion === undefined || deployment.awsRegion === ""){
+                    deploymentError.push("AWS account id cannot be empty");
+                }
+            } else if (deployment.cloudProvider === "azure") {
+                if(deployment.subscriptionId === undefined || deployment.subscriptionId === ""){
+                    deploymentError.push("Azure subscription id cannot be empty");
+                }
+                if(deployment.tenantId === undefined || deployment.tenantId === ""){
+                    deploymentError.push("Azure tenant id cannot be empty");
+                }
+            }
+        }
+        if (deployment.deploymentType === undefined || deployment.deploymentType === "") {
+            deploymentError.push("Deployment Type cannot be empty");
+        }
+        if (deployment.clusterName === undefined || deployment.clusterName === "") {
+            deploymentError.push("Cluster Name cannot be empty");
+        }
+        if (deployment.ingressDomain === undefined || deployment.ingressDomain === "") {
+            deploymentError.push("Ingress Domain cannot be empty");
+        }
+        if (deployment.ingressType === undefined || deployment.ingressType === "") {
+            deploymentError.push("Ingress Type cannot be empty");
+        }
+        if (deployment.kubernetesUseDynamicStorage !== undefined && deployment.kubernetesUseDynamicStorage === true
+            && (deployment.kubernetesStorageClassName === undefined || deployment.kubernetesStorageClassName === ""
+                || deployment.kubernetesStorageProvisioner === undefined || deployment.kubernetesStorageProvisioner === "")) {
+            deploymentError.push("Storage Provisioner/Storage Class Name cannot be empty");
+        }
+
+        // return error response
+        if (deploymentError.length > 0) {
+            console.log(deploymentError);
+            return res.status(406).send(deploymentError);
```

Comment:
400 Bad Request

Reply:
like empty string/undefined value should not be accepted right ? so is 406(Not acceptable) not appropriate ?